### PR TITLE
external/webserver: Add support of large body in response

### PR DESCRIPTION
**Pull request by amitpurwar**
This adds support for webserver to send the response body
larger than HTTP_CONF_MAX_REQUEST_LENGTH bytes

Signed-off-by: Amit Purwar <amit.purwar@samsung.com>

### CODE REVIEW

File: external/webserver/http_client.c [hunk 1]
```diff
@@ -1163,36 +1201,26 @@ int http_send_response_helper(struct http_client_t *client, int status, const ch
...
-		if (ret < 1) {
-			HTTP_FREE(buf);
-			return HTTP_ERROR;
-		} else {
-			sndlen -= ret;
-			buflen += ret;
-		}
+	//send remaining body
+	if (rem_body_len > 0) {
+		http_send_buffer(client, body + len, rem_body_len);
```
Comment (sangam-swami) [inline external/webserver/http_client.c:1221]: return type check is not present

File: external/webserver/http_client.c [hunk 2]
```diff
@@ -1059,12 +1059,48 @@ int http_send_response_chunk(struct http_client_t *client, int status, const cha
...
+		if (client->server->tls_init) {
+			ret = mbedtls_ssl_write(&(client->tls_ssl), (unsigned char *)buf + buflen, sndlen);
+		} else
+#endif
+		{
+			ret = send(client->client_fd, buf + buflen, sndlen, 0);
+		}
+
+		if (ret < 1) {
+			HTTP_LOGE("Fail to send buffer ret[%d] \n", ret);
```
Comment (jeongchanKim) [inline external/webserver/http_client.c:1085]: I think it is better to print the errno also.

File: external/webserver/http_client.c [hunk 3]
```diff
@@ -1059,12 +1059,48 @@ int http_send_response_chunk(struct http_client_t *client, int status, const cha
...
+#ifdef CONFIG_NET_SECURITY_TLS
+		if (client->server->tls_init) {
+			ret = mbedtls_ssl_write(&(client->tls_ssl), (unsigned char *)buf + buflen, sndlen);
+		} else
+#endif
+		{
+			ret = send(client->client_fd, buf + buflen, sndlen, 0);
+		}
+
+		if (ret < 1) {
```
Comment (jeongchanKim) [inline external/webserver/http_client.c:1084]: Maybe `ret` is successfully sent bytes.
How about renaming `ret` like `send_bytes`or something much better to understand?

File: external/webserver/http_client.c [hunk 4]
```diff
@@ -1163,36 +1201,26 @@ int http_send_response_helper(struct http_client_t *client, int status, const ch
...
 		buflen += snprintf(buf + buflen,
 					HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "\r\n");
-
 		// Include response body
 		if (body) {
-			buflen += snprintf(buf + buflen,
-						HTTP_CONF_MAX_REQUEST_LENGTH - buflen,
-						"%s",
-						body);
+			memcpy(buf + buflen, body, HTTP_CONF_MAX_REQUEST_LENGTH - buflen);
```
Comment (jeongchanKim) [inline external/webserver/http_client.c:1206]: How about declaring a local variable and set as `HTTP_CONF_MAX_REQUEST_LENGTH - buflen`?
`HTTP_CONF_MAX_REQUEST_LENGTH - buflen` is used multiple times, so we can reduce the code and improve the performance.

File: external/webserver/http_client.c (hunk 1 above)
Comment (amitpurwar) [reply external/webserver/http_client.c:1221]: done

File: external/webserver/http_client.c (hunk 2 above)
Comment (amitpurwar) [reply external/webserver/http_client.c:1085]: done

File: external/webserver/http_client.c (hunk 3 above)
Comment (amitpurwar) [reply external/webserver/http_client.c:1084]: done

File: external/webserver/http_client.c (hunk 4 above)
Comment (amitpurwar) [reply external/webserver/http_client.c:1206]: not required

**Comment by vibhor-m** [comment]
Changes look fine and as per change requested

File: external/webserver/http_client.c [hunk 5]
```diff
@@ -1059,12 +1059,49 @@ int http_send_response_chunk(struct http_client_t *client, int status, const cha
...
+{
+	int send_byte = 0;
+	int sndlen = 0;
+	int buflen = 0;
+
+	if (client == NULL) {
+		HTTP_LOGE("client is NULL \n");
+		return -1;
+	}
+
```
Comment (j11591) [inline external/webserver/http_client.c:1072]: How about checking const char *buf?

Comment (amitpurwar) [reply external/webserver/http_client.c:1072]: done

**Review by j11591 — APPROVED** [review]
LGTM

**Pull request merged**